pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -79,7 +82,21 @@ int getPriority(BaseFilterOperator filterOperator) {
           return 3;
         }
         if (filterOperator instanceof ScanBasedFilterOperator) {
-          return 4;
+          DataSourceMetadata metadata = filterOperator.getDataSourceMetadata();
+          int basePriority = 4;
+          if (metadata == null) {
+            // shouldn't happen
+            return basePriority;
+          }
+          // lower priority for multivalue columns
+          if (! metadata.isSingleValue()) {
```

Comment:
Please format the changes before committing (referring to: ! metadata)

Reply:
Done.